### DISCUSSION TITLE
first post, new to templater, what will I need to learn to write this conditional?

### DISCUSSION THREAD

**Discussion by thor314** [SilentVoid13/Templater#732] [Help]
I have a snippet I'd like to evaluate in my daily note:
```js
<%*
    let day = tp.date.now("dddd");
    let isWeekday = (day == "Monday" || day == "Tuesday" || day == "Wednesday" || day == "Thursday" || day == "Friday" );
    if isWeekday {
        tR += "something related to weekdays"
    }
%>
```
When I try to evaluate it, I see template parsing error. I don't know where to look to get more info about this error. I'm also not a frequent js user.

I've been looking thru:
- [templater docs](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/date-module.html#tpdatenowformat-string--yyyy-mm-dd-offset-numberstring-reference-string-reference_format-string)
- [w3 schools docs](https://www.w3schools.com/jsref/jsref_if.asp) for info about js
- [eta docs](https://eta.js.org/docs/syntax) for info about eta

But I think my issue mostly involves not having a good workflow around writing js and finding errors/bugs. Looking for resources and info about same.

**Comment by AB1908** [comment]
```js
<%*
    let day = tp.date.now("dddd");
    if (day == "Monday" || day == "Tuesday" || day == "Wednesday" || day == "Thursday" || day == "Friday" ) {
        tR += "something related to weekdays"
    }
%>
```

**Reply by thor314** [reply]
The question was more a "how do I use this tool" question, than a "please fix my problem" question, but thanks!

**Reply by AB1908** [reply]
Your logic was wrong. Try your original second statement in the console to see what it actually stores. You'll get there with a bit of practice!

**Reply by thor314** [reply]
Is there a guide somewhere on how to use the console? I'm not too familiar, here's what I've tried to do:

```
    let day = tp.date.now("dddd");
VM330:1 Uncaught ReferenceError: tp is not defined
    at <anonymous>:1:15
(anonymous) @ VM330:1
index.html:1 Uncaught (in promise) DOMException: Document is not focused.
Promise.then (async)
VimrcPlugin.captureYankBuffer @ plugin:obsidian-vimrc-support:850
eval @ plugin:obsidian-vimrc-support:454
<%*
    let day = tp.date.now("dddd");
%>
VM338:1 Uncaught SyntaxError: Unexpected token '<'
<% let day = tp.date.now("dddd");
VM438:1 Uncaught SyntaxError: Unexpected token '<'
<% let day = tp.date.now("dddd"); %>
VM448:1 Uncaught SyntaxError: Unexpected token '<'
```

**Reply by AB1908** [reply]
Hmm I don't know how to put this into words but think of the console as a place to run a statement individually to see what it does. Maybe look up how other folks use it on YouTube. I'm happy to show you on Discord if you DM me. I usually hang out on the Obsidian Server.
